lib/screens/v2/screen_data.ex

```diff
@@ -31,6 +32,12 @@ defmodule Screens.V2.ScreenData do
     select_variant(screen_id, opts, &layout_to_data/2)
   end
 
+  @spec get_with_flex_zone(screen_id()) :: data_with_flex_zone()
+  @spec get_with_flex_zone(screen_id(), options()) :: data_with_flex_zone()
+  def get_with_flex_zone(screen_id, opts \\ []) do
```

Comment:
This essentially is the same thing as `simulation`, it just returns the data in a slightly different format. It might be simpler to leave this module unchanged, call `simulation` from the controller in the `mercury` special case, and just reformat the data there. Like

```elixir
%{full_page: data, flex_zone: flex_zone} = ScreenData.simulation(...)
%{@base_response | data: data, flex_zone: flex_zone}
```

This makes the function name `simulation`, and the specific format of `simulation_data()`, a bit awkward because this is no longer specific to Screenplay simulations. I would be fine letting this awkwardness sit, since we'd expect to revisit it as we think more about client-side versus server-side paging.

Reply:
Wow, totally overlooked that in my first pass. I like that much better.